packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -35,3 +60,424 @@ def __init__(self, **kwargs):
         :param performative: the type of message.
         """
         super().__init__(**kwargs)
+        self.w3 = Web3(Web3.HTTPProvider(PROVIDER))
+        self.address = None
+        self.deployed = False
+        self.abi = None
+        self.bytecode = None
+        self.instance = None
+        batch_token_ids = []
```

Comment:
this is very verbose, can't the helper method return a list of item_ids?

Reply:
As I say in the comment I can actually remove the token_id from there and pass it only when the transaction happens. But this is something that can remove once I have all the functionality.

Reply:
well I think its helpful to actually have access to the token ids here. where else would you store them?

Reply:
I would just generate them in the skill and pass them to the contract. Then we should create an other method inside the contract that returns the ids. It makes sense to have access to the ids if you are the one who deployed the contract but if you are just a user you have no idea what the contracts contains. With that being said the user must be able to fetch the ids from the contract it self.

Reply:
Well I think the methods to create them should be attached to the contract but callable from the skill, potentially with some optional ids to be fed in from the skill